presto-parquet/src/main/java/io/prestosql/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -102,6 +103,8 @@ public PrimitiveColumnWriter(Type type, ColumnDescriptor columnDescriptor, Primi
         this.compressionCodec = requireNonNull(compressionCodecName, "compressionCodecName is null");
         this.compressor = getCompressor(compressionCodecName);
         this.pageSizeThreshold = pageSizeThreshold;
+
+        this.columnStatistics = Statistics.createStats(columnDescriptor.getPrimitiveType());
```

Comment:
Are there statistics for non-primitive types?

Reply:
No. This aims to fill the stats part for primitive columns https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L740 . There is no support for this on non-primitive types.